Assets/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -739,6 +739,42 @@ public void Actions_CanReadValueFromAction_InCallback()
         Assert.That(receivedValue, Is.EqualTo(1).Within(0.00001));
     }
 
+    [Test]
+    [Category("Actions")]
+    public void Actions_CanReadValueFromAction_InCallback_AsButton()
+    {
+        InputSystem.settings.defaultButtonPressPoint = 0.5f;
+
+        var action = new InputAction(binding: "<Gamepad>/leftTrigger");
+        var gamepad = InputSystem.AddDevice<Gamepad>();
+
+        action.Enable();
+
+        bool? receivedValue = null;
+        action.performed +=
+            ctx =>
+        {
+            Assert.That(receivedValue, Is.Null);
+            receivedValue = ctx.ReadValueAsButton();
+        };
+
+        Set(gamepad.leftTrigger, 0.25f);
+
+        Assert.That(receivedValue, Is.False);
```

Comment:
Not a fan of test assert messages TBH (unlike for asserts in normal code). It adds a *shit ton* of clutter to the tests and makes them extremely wordy while IMO adding not that much in value. If an assert fails, I'll have to go and usually debug the tests anyway to find what I've broken. Having the message wouldn't speed up the process. If the test is well-written, I will very quickly see why the expectation is the way it is.

But... maybe it's all lame excuses and I'm just lazy. Not saying this might not be the case :)

Reply:
Haha no problem